### DISCUSSION TITLE
Git on desktop is asking for GitHub password even though I have SSH set up

### DISCUSSION THREAD

**Discussion by novdy** [community/community#23582] [New to GitHub]
<p>Hello, I’m running into a major issue on a new machine. I’ve followed the GitHub documentation on how to set up an SSH key and have tested it using “ssh -T <a href="mailto:git@github.com">git@github.com</a>” and get the correct message</p>
<p>“Hi novdy! You’ve successfully authenticated, but GitHub does not provide shell access.”</p>
<p>But when I try and run “git push -u origin main”, it starts asking for my GitHub credentials. Once entered, it doesn’t accept them since of course, GitHub doesn’t allow password authentication for this anymore.</p>
<p>I did not set an SSH passphrase. I wouldn’t imagine an authentication token being necessary.</p>
<p>If there is any way you all could help, I’d really appreciate it!</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Probably your repository is still configured with HTTPS URLs for the GitHub repository. You can check with:</p>
<pre><code class="lang-auto">git remote show -n origin
</code></pre>
<p>If you see HTTPS URLs, you can use <a href="https://git-scm.com/docs/git-remote#Documentation/git-remote.txt-emset-urlem"><code>git remote set-url</code></a> to change them to the matching SSH URLs to make Git use SSH. Otherwise I’m curious what it looks like. :smile_cat:</p>

**Comment by novdy** [comment]
<p>Thanks so much for this! I knew it had to be something simple like this. :pray:</p>
